fix: Following "next" URL after login.

suiting-young opened the pull request:
<!--- Thank you for contributing to Flask-Appbuilder. -->
<!--- This repo uses a PR lint bot (https://github.com/apps/prlint), make sure to prefix your PR title with one of: -->
<!--- build|chore|ci|docs|feat|fix|perf|refactor|style|test|other -->

### Description

The `has_access` decorator provides a "next" parameter when redirecting to this endpoint.
https://github.com/dpgaspar/Flask-AppBuilder/blob/b9b3259a9d47a14430144c8362215c2143f06aa4/flask_appbuilder/security/decorators.py#L117-L122
Should try to redirect back here if login succeeded.

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Is CRUD MVC related.
- [x] Is Auth, RBAC security related.
- [ ] Changes the security db schema.
- [ ] Introduces new feature
- [ ] Removes existing feature

dpgaspar reviewed (commented):
looks good, can you add some tests

suiting-young commented:
I have added a parent class `AuthBasicView` for both `AuthDBView` and `AuthLDAPView`.
As the `login` method of them are extremely same, except the auth method.
And furthermore, I don't know how to test `AuthLDAPView` 😝 . Codecov tells me that my test is against `AuthDBView`.

---
_P.S._ I removed `from urllib(.parse) import quote` Python2 fallback from `test_mvc.py`, because other 5 files only use Python3 module structure.

suiting-young commented:
@dpgaspar could you please review again, thanks.

suiting-young commented:
Hi @dpgaspar,

I have rebased the PR to latest `master`.
And there is only one thing left that need to be fixed then:

If the user failed to login,
- it should stay at the **current** page (_**login** actually_),
  with any parameter (_e.g. `?next=...`_) retained;
- instead of redirecting to purified `get_url_for_login`.

Please have a look.
Thanks
